src/containerapp/azext_containerapp/custom.py

```diff
@@ -4297,3 +4306,195 @@ def delete_workload_profile(cmd, resource_group_name, env_name, workload_profile
+        [executor.submit(patch_get_image_inspection, pack_exec_path, img, inspect_results) for img in imgs]
+
+    # Get the current tags of Dotnet Mariners
+    oryx_run_img_tags = get_current_mariner_tags()
+    failed_reason = "Failed to inspect the image. Please make sure that you are authenticated to the container registry and that the image exists."
+    not_based_mariner_reason = "Image not based on Mariner"
+    mcr_check_reason = "Image not from mcr.microsoft.com/oryx/builder"
+    results = []
+    # Start checking if the images are based on Mariner
+    print("\rChecking for patches...", end="", flush=True)
+    for inspect_result in inspect_results:
+        if inspect_result["remote_info"] == 401:
+            results.append(dict(targetContainerName=inspect_result["targetContainerName"], targetContainerAppName=inspect_result["targetContainerAppName"], targetContainerAppEnvironmentName=inspect_result["targetContainerAppEnvironmentName"], targetResourceGroup=inspect_result["targetResourceGroup"], targetImageName=inspect_result["image_name"], oldRunImage=None, newRunImage=None, id=None, reason=failed_reason))
+        else:
+            # Divide run-images into different parts by "/"
+            run_images_props = inspect_result["remote_info"]["run_images"]
+            if run_images_props is None:
+                results.append(dict(targetContainerName=inspect_result["targetContainerName"], targetContainerAppName=inspect_result["targetContainerAppName"], targetContainerAppEnvironmentName=inspect_result["targetContainerAppEnvironmentName"], targetResourceGroup=inspect_result["targetResourceGroup"], targetImageName=inspect_result["image_name"], oldRunImage=None, newRunImage=None, id=None, reason=not_based_mariner_reason))
+            else:
+                for run_images_prop in run_images_props:
+                    if run_images_prop["name"].find("mcr.microsoft.com/oryx/builder") != -1:
+                        run_images_prop = run_images_prop["name"].split(":")
+                        run_images_tag = run_images_prop[1]
+                        # Based on Mariners
+                        if run_images_tag.find('mariner') != -1:
+                            check_result = patchable_check(run_images_tag, oryx_run_img_tags, inspect_result=inspect_result)
+                            results.append(check_result)
+                        else:
+                            results.append(dict(targetContainerName=inspect_result["targetContainerName"], targetContainerAppName=inspect_result["targetContainerAppName"], targetContainerAppEnvironmentName=inspect_result["targetContainerAppEnvironmentName"], targetResourceGroup=inspect_result["targetResourceGroup"], targetImageName=inspect_result["image_name"], oldRunImage=inspect_result["remote_info"]["run_images"]["name"], newRunImage=None, id=None, reason=failed_reason))
+                    else:
+                        # Not based on image from mcr.microsoft.com/dotnet
```

Comment:
@snehapar9 @harryli0108 -- this message does not match up with `mcr_check_reason`

Reply:
Updated: modified reason to match with the scenario